cognite_toolkit/_cdf_tk/client/data_classes/charts_data.py

```diff
@@ -0,0 +1,303 @@
+import sys
+from dataclasses import dataclass, field, fields
+from functools import lru_cache
+from typing import Any
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes._base import CogniteObject
+from cognite.client.data_classes.data_modeling import NodeId, ViewId
+from cognite.client.utils._auxiliary import to_camel_case
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+@dataclass
+class ChartObject(CogniteObject):
+    # ChartObjects are used in the frontend and the backend does not do any validation of these fields.
+    # Therefore, to ensure that we do not lose any data, we store unknown fields in a separate dictionary.
+    # This allows unknown fields to be preserved when loading and dumping ChartObjects
+    # (serialization and deserialization).
+    _unknown_fields: dict[str, object] | None = field(default=None, init=False, repr=False)
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        """Load a ChartObject from a dictionary."""
+        instance = super()._load(resource, cognite_client=cognite_client)
+        known_camel_case_props = cls._known_camel_case_props()
+        instance._unknown_fields = {k: v for k, v in resource.items() if k not in known_camel_case_props}
+        return instance
+
+    @classmethod
+    @lru_cache(maxsize=1)
```

Comment:
This deserves a docstring or comment explaining why it is safe to cache on the class [because fields are presumed immutable for a class], and we're not going to support dynamic programming that violates that expectation.